### ISSUE TITLE
Generating protos in sub directories fails

### ISSUE THREAD

**Issue by yohcop** [bazelbuild/rules_go#173]
I'm trying to use a vendoring/third_party library with a flat BUILD file. The problem I have is that it contains a proto file deep in the tree. It seems fine to create a go_library with source files in sub directories, but protos don't seem to work.

I've made up a simple example. Something like:

```
~/tmp/go-bzl % tree lib 
lib
├── BUILD
└── sub
    └── bar.proto
```

```
~/tmp/go-bzl % cat lib/BUILD                                   
load("@io_bazel_rules_go//go:def.bzl", "go_library")
load("@io_bazel_rules_go//proto:go_proto_library.bzl", "go_proto_library")

go_proto_library(
    name = "sub/bar_proto",  # Also tried name="bar_proto"
    srcs = [
        "sub/bar.proto",
    ],
    visibility = ["//visibility:public"],
)
```

```
~/tmp/go-bzl % bazel build lib:sub/bar_proto                
INFO: Found 1 target...
ERROR: /home/y/tmp/go-bzl/lib/BUILD:20:1: Generating into bazel-out/local-fastbuild/bin/lib/sub/bar_proto/sub failed: linux-sandbox failed: error executing command /home/y/.cache/bazel/_bazel_y/1129aeaeb86f4311cfe08c57d10b52b4/execroot/go-bzl/_bin/linux-sandbox ... (remaining 6 argument(s) skipped).
bazel-out/local-fastbuild/bin/lib/sub/: No such file or directory
Target //lib:sub/bar_proto failed to build
```

I also inspected the sandbox, and it seems that just like the error says, there is no directory bazel-out/local-fastbuild/bin/lib/sub/, and protoc probably just fails when trying to write the proto. Should go_proto_library create the directories for output files? Any other ways to make this work?

Thanks,

**Comment by pmbethe09** [comment]
Probably a bug.  go_proto_library was never tested for the flattened BUILD
file use-case, so I am not surprised that it is not working.

**Comment by yohcop** [comment]
Here is another "bug" (maybe), that seems related.

Since the flat structure was getting complicated, I tried a completely different approach: a repo in //external.

I made the sub-project a standalone, bazel-based, project. Made sure everything was building.
Then I added it as an external repo (in my case using `local_repository`). It has its own `WORKSPACE`, and `BUILD` with `go_prefix`. And this repo has a `go_proto_library` in it.

Trying to build //external/github.com/foo/lib:my_proto fails, with the .pb.go file being not generated (I suspect they are, just not in the right directory). I can try to make a simple repeatable example if needed.
